packages/rum-react/test/specs/get-apm-routes.spec.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+/**
+ * shims are required for enzyme to work on older browsers.
+ */
+
+import 'airbnb-js-shims/target/es2015'
+import React from 'react'
+import Enzyme, { mount } from 'enzyme'
+import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+import {
+  MemoryRouter as Router,
+  Route,
+  useNavigate,
+  useLocation,
+  Navigate
+} from 'react-router-dom'
+import { ApmBase } from '@elastic/apm-rum'
+import {
+  createServiceFactory,
+  TRANSACTION_SERVICE
+} from '@elastic/apm-rum-core'
+import { getApmRoutes } from '../../src/get-apm-routes'
+import { getGlobalConfig } from '../../../../dev-utils/test-config'
+
+let navigate
+function Component(props) {
+  // React-router util which allows navigating between routes
+  // Cleaner than handling the history api manually
+  navigate = useNavigate()
```

Comment:
Does  it have to be inside the component?

Reply:
yes, you can't use an effect outside a component